buildtools/commands/update-readme.js

```diff
@@ -0,0 +1,27 @@
+const { resolve, join } = require('path');
+const shell = require('shelljs');
+const replace = require('replace-in-file');
+
+const ROOT_DIR = resolve(__dirname, '../../');
+
+function getPublishedWidgetVersion () {
+  const stdout = shell.exec('yarn info @okta/okta-signin-widget versions', { silent: true });
+  const arrayStr = stdout.substring(stdout.indexOf('['), stdout.lastIndexOf(']') + 1).replace(/'/g, '"');
+  const versions = JSON.parse(arrayStr);
+  const version = versions[versions.length - 1];
+  console.log('Last published okta-signin-widget version: ', version);
+  return version;
+}
+
+exports.command = 'update-readme';
+exports.describe = 'Updates CDN URLs in the README file';
+exports.handler = () => {
+  const version = getPublishedWidgetVersion();
```

Comment:
Been awhile since I've dug into the publish code, but why can't we reference the `package.json` version?

Since this build tool is intended to hook into our promotion flow, I'd assume that when the artifact is promoted the `version` can be treated as our source of truth.

This would also mitigate any delays in `npm publish` where the artifact may not be present at the time this script is executed.

Reply:
It would be nice to fit this into our promotion flow, but we also need to be able to manually update it at anytime. 

The package.json can only be considered a "source of truth" for this CDN url during a very narrow of slice of time, which you identified: after promote, but before version bump. And at that time the conductor release train has not been done yet so technically the URL would be invalid. If we wired it up to execute at that stage, then I think it would make sense for the script to accept a widget version as a command line argument. If the caller has a superior source of truth then it may provide the value.

Since the README update should link to valid URLs, it probably makes more sense to wire it up in the conductor release train so it runs after the CDN deployment. We will have to be careful if it is passing an argument and overriding the "latest" dist tag from NPM because sometimes we often release trains to patch older versions, and these should not update the README (or the "latest" dist tag).